### PR TITLE
[PowerPC] fix legalization crash

### PR DESCRIPTION
**Pull request by RolandF77**
If v2i64 scalar_to_vector is made custom, llc can crash in certain legalization cases where v2i64 vectors are injected, even if they weren't otherwise present. The code generated would be fine, but that operation is not handled in ReplaceNodeResults. Add handling.

### CODE REVIEW

File: llvm/test/CodeGen/PowerPC/custom-stov.ll [hunk 1]
```diff
@@ -0,0 +1,60 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -verify-machineinstrs < %s | FileCheck %s
+
+target datalayout = "E-m:e-Fi64-i64:64-n32:64-S128-v256:256:256-v512:512:512"
+target triple = "powerpc64-unknown-linux-gnu"
```
Comment (chenzheng1030) [inline llvm/test/CodeGen/PowerPC/custom-stov.ll:5]: nit: Can we add this to the RUN line?

File: llvm/test/CodeGen/PowerPC/custom-stov.ll [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  ret void
+
+for.cond.cleanup56:                               ; preds = %entry
+  ret void
+}
+
+; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none)
+declare i32 @llvm.vector.reduce.umax.v4i32(<4 x i32>) #1
+
+attributes #0 = { "target-cpu"="ppc64" }
```
Comment (chenzheng1030) [inline llvm/test/CodeGen/PowerPC/custom-stov.ll:59]: nit: can we add this to the RUN line?

File: llvm/test/CodeGen/PowerPC/custom-stov.ll [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+attributes #1 = { nocallback nofree nosync nounwind speculatable willreturn memory(none) }
```
Comment (chenzheng1030) [inline llvm/test/CodeGen/PowerPC/custom-stov.ll:60]: nit: it this necessary to reproduce this crash?

**Review by chenzheng1030 — APPROVED** [review]
Looks reasonable to me except some nits in the test.

File: llvm/test/CodeGen/PowerPC/custom-stov.ll [hunk 4]
```diff
@@ -0,0 +1,60 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -verify-machineinstrs < %s | FileCheck %s
```
Comment (amy-kwan) [inline llvm/test/CodeGen/PowerPC/custom-stov.ll:2]: Can we also add `-ppc-asm-full-reg-names -ppc-vsr-nums-as-vr` to the RUN line?

**Review by amy-kwan — APPROVED** [review]
Aside from the nits on the test, it also LGTM.

File: llvm/test/CodeGen/PowerPC/custom-stov.ll [hunk 5]
```diff
@@ -0,0 +1,53 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -mcpu=ppc64 -mtriple=powerpc64-unknown-linux-gnu -verify-machineinstrs -ppc-asm-full-reg-names -ppc-vsr-nums-as-vr < %s | FileCheck %s
```
Comment (amy-kwan) [inline llvm/test/CodeGen/PowerPC/custom-stov.ll:2]: One final nit: The line is a bit too long. Can we separate the line?

**Review by amy-kwan — APPROVED** [review]
Final LGTM.

**Pull request merged**